title-updater.js

```
@@ -0,0 +1,103 @@
+        });
+    }
+
+    // Watch for title changes using MutationObserver
+    if (typeof MutationObserver !== 'undefined') {
+        const observer = new MutationObserver(function (mutations) {
+            mutations.forEach(function (mutation) {
+                if (mutation.type === 'childList' || mutation.type === 'characterData') {
+                    // Check if title-related elements changed
+                    const titleChanged = Array.from(mutation.addedNodes).some(node =>
+                        node.nodeType === Node.ELEMENT_NODE &&
+                        (node.tagName === 'H1' || node.querySelector && node.querySelector('h1'))
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)